src/storage/rdb.cc

```diff
@@ -680,3 +679,323 @@ Status RDB::LoadRdb(uint32_t db_index, bool overwrite_exist_key) {
 
   return Status::OK();
 }
+
+Status RDB::Dump(const std::string &key, const RedisType type) {
+  unsigned char buf[2];
+  /* Serialize the object in an RDB-like format. It consist of an object type
+   * byte followed by the serialized object. This is understood by RESTORE. */
+  auto s = SaveObjectType(type);
+  if (!s.IsOK()) return {Status::RedisExecErr, s.Msg()};
+  s = SaveObject(key, type);
+  if (!s.IsOK()) return {Status::RedisExecErr, s.Msg()};
+
+  /* Write the footer, this is how it looks like:
+   * ----------------+---------------------+---------------+
+   * ... RDB payload | 2 bytes RDB version | 8 bytes CRC64 |
+   * ----------------+---------------------+---------------+
+   * RDB version and CRC are both in little endian.
+   */
+
+  /* RDB version */
+  buf[0] = MaxRDBVersion & 0xff;
+  buf[1] = (MaxRDBVersion >> 8) & 0xff;
+  s = stream_->Write((const char *)buf, 2);
+  if (!s.IsOK()) {
+    return {Status::RedisExecErr, s.Msg()};
+  }
+
+  /* CRC64 */
+  std::string &output = static_cast<RdbStringStream *>(stream_.get())->GetInput();
```

Comment:
If you want to assume it must be an `RdbStringStream` here, you can add a `CHECK` statement.

Reply:
But TBH, to support output files rather than strings, we need a better solution here.

Reply:
For the Redis dump command, it needs to echo the result to the client, so it should be good to keep in memory. Not sure if I misunderstood @PragmaTwice point.

Reply:
> If you want to assume it must be an `RdbStringStream` here, you can add a `CHECK` statement.

I added a member attribute named type_ to RdbStream, and I ensure its safety by checking it before performing a static_cast